src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -220,6 +258,150 @@ private void triggerCheckpoint(String checkpointIdentifier) throws FlinkExceptio
         checkpointTrigger.triggerCheckpoint(checkpointId);
     }
 
+    // ------------------------------------------------------------------------
+    //  metrics
+    // ------------------------------------------------------------------------
+
+    /**
+     * Gauge for getting the unread bytes information from reader group.
+     */
+    private static class UnreadBytesGauge implements Gauge<Long> {
+
+        private final ReaderGroup readerGroup;
+
+        public UnreadBytesGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public Long getValue() {
+            return readerGroup.getMetrics().unreadBytes();
+        }
+    }
+
+    /**
+     * Gauge for getting the reader group name information from reader group.
+     */
+    private static class ReaderGroupNameGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+
+        public ReaderGroupNameGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getGroupName();
+        }
+    }
+
+    /**
+     * Gauge for getting the scope name information from reader group.
+     */
+    private static class ScopeNameGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+
+        public ScopeNameGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getScope();
+        }
+    }
+
+    /**
+     * Gauge for getting online readers information from reader group.
+     */
+    private static class OnlineReadersGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String separator = ",";
+
+        public OnlineReadersGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getOnlineReaders().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting stream name information from reader group.
+     */
+    private static class StreamNamesGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String separator = ",";
+
+        public StreamNamesGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getStreamNames().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting position information of each segment of a stream from reader group.
+     */
+    private static class SegmentPositionsGauge implements Gauge<String> {
```

Comment:
I haven't tested for multiple streams but I realized that I am filtering only on the stream name instead of stream and scope. Will update the code and test with multiple streams.